pkgs/tools/networking/airgeddon/default.nix

```diff
@@ -0,0 +1,126 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+  # Required
+, aircrack-ng
+, bash
+, gawk
+, iproute2
+, iw
+, pciutils
+, procps
+, tmux
+  # Optionals
+  # Missing in nixpkgs: beef, hostapd-wpe, asleap
+, bully
+, crunch
+, dhcp
+, dnsmasq
+, ettercap
+, hashcat
+, hcxdumptool
+, hcxtools
+, hostapd
+, lighttpd
+, mdk4
+, nftables
+, openssl
+, pixiewps
+, reaverwps-t6x # Could be the upstream version too
+, wireshark-cli
+  # Support groups
+, supportWpaWps ? true # Most common use-case
+, supportHashCracking ? false
+, supportEvilTwin ? false
+}:
+with lib;
+let
+  deps = [
+    aircrack-ng
+    bash
+    gawk
+    iproute2
+    iw
+    pciutils
+    procps
+    tmux
+  ] ++ (lists.optionals supportWpaWps [
+    bully
+    pixiewps
+    reaverwps-t6x
+  ]) ++ (lists.optionals supportHashCracking [
+    crunch
+    hashcat
+    hcxdumptool
+    hcxtools
+    wireshark-cli
+  ]) ++ (lists.optionals supportEvilTwin [
+    dhcp
+    dnsmasq
+    ettercap
+    hostapd
+    lighttpd
+    openssl
+    mdk4
+    nftables
+  ]);
+in
+stdenv.mkDerivation rec {
+  pname = "airgeddon";
+  version = "11.01";
+
+  src = fetchFromGitHub {
+    owner = "v1s1t0r1sh3r3";
+    repo = "airgeddon";
+    rev = "v${version}";
+    sha256 = "3TjaLEcerRk69Ys4kj7vOMCRUd0ifFJzL4MB5ifoK68=";
+  };
+
+  strictDeps = true;
+  buildInputs = [ makeWrapper ];
+
+
+  # What these replacings do?
+  # - Disable the auto-updates (we'll run from a read-only directory);
+  # - Silence the checks (NixOS will enforce the PATH, it will only see the tools as we listed);
+  # - Use "tmux", we're not patching XTerm commands;
+  # - Remove PWD and $0 references, forcing it to use the paths from store;
+  # - Force our PATH to all tmux sessions.
+  postPatch = ''
+    patchShebangs airgeddon.sh
+    sed -i '
+      s|AIRGEDDON_AUTO_UPDATE=true|AIRGEDDON_AUTO_UPDATE=false|
+      s|AIRGEDDON_SILENT_CHECKS=false|AIRGEDDON_SILENT_CHECKS=true|
+      s|AIRGEDDON_WINDOWS_HANDLING=xterm|AIRGEDDON_WINDOWS_HANDLING=tmux|
+      ' .airgeddonrc
+
+    sed -Ei '
+      s|\$\(pwd\)|${placeholder "out"}/share/airgeddon;scriptfolder=${placeholder "out"}/share/airgeddon/|
+      s|\$\{0\}|${placeholder "out"}/bin/airgeddon|
+      s|tmux send-keys -t "([^"]+)" "|tmux send-keys -t "\1" "export PATH=\\"$PATH\\"; |
+      ' airgeddon.sh
+  '';
```

Comment:
This does looks like a good use-case for `resholve.mkDerivation`: https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/misc/shunit2/default.nix

Reply:
I did try, and it helped me find many non-listed dependencies. But this is beyond a simple packager's job. This script generates other scripts in `/tmp` during runtime and launches them in nested tmux/xterm sessions.

Attaching the effort it took for it to launch, in the hope it may help find more dependencies in later updates: https://gist.github.com/PedroHLC/e6cefa4dce6474d27f3a91f4ff929a67

Reply:
CC @abathur. You may want to take a look at this.

Reply:
Hehe. I keep a search open, so I'd noticed your comment this morning. I noticed how large airgeddon was and was going to be a bit surprised if there wasn't a blocker in there somewhere.

I assume it'll just make a good addition to the list of ~white-whales, but I'll try to find time to poke at it tonight in case there's something that I know a way past.

@PedroHLC thanks for posting the gist :)

Reply:
I gave this a try myself tonight and have 3 different thoughts:
1. Thanks for giving resholve a really thorough try, @PedroHLC. This is one of the most ambitious packages a first-time resholve user has attempted, so I'm curious how well the documentation worked. (Feel free to ignore this--I don't really expect more of your time. But, I'd also be happy to chat about it over Matrix, or via the email in my profile?)
2. I think abathur/resholve#78 may make a few bits of this a little cleaner, so I'm inclined to hold up for that before giving it another try. I'll bump it up my priority list.
3. > @PedroHLC: This script generates other scripts in /tmp during runtime and launches them in nested tmux/xterm sessions.

    I'm curious if resholve was causing a specific problem here, and if so--what it looked like?

Reply:
I'll give you a step-for-step introduction of what I did and what I found, maybe this can help you to help me:

I first took a look at Kokadas' derivation and saw that it was primarily rounded on that "solutions" entry after adding the "mkDerivation". So I searched how to write one of those, which led me to [this README](https://github.com/nixos/nixpkgs/blob/master/pkgs/development/misc/resholve/README.md) that I've used as the source of truth.

My first doubt was, "Should I use the paths relative to input or output?" and well, that was written there, "$out-relative paths".

The second was an observation that I've seen in Kokadas' code. He was using "sh" instead of "bash" as interpreter, which led me to ask him and then discovering bash [includes a posix-mode](https://www.gnu.org/software/bash/manual/html_node/Bash-POSIX-Mode.html).

Then a small try after just adding the "scripts", "interpreter", and the "inputs" I've already known lead me to that regex parsing issue of oilshell, which I solved with the sed `s|^(.+) =~ ([^\$].+) ]]|regexp='\2'; \1 =~ $regexp ]]|` that I based in a commit from you in 2021.

After that I had to deal with a `Can't resolve dynamic argument in 'source'` -- that's because airgeddon sources its scripts with `source "${scriptfolder}${rc_file_name}"` and well, I don't know how to correctly solve these even now :see_no_evil: -- Should I replace this with relative-hardcoded-paths, I know that I can't use absolute-hardcoded-paths -- Kokada pointed me to "`keep.source`" but this didn't help, "`fix`" somehow did the trick for the "strings" and "known_pins", and for the "plugins" in a little hacky way. But I had to leave ".airgeddonrc" out of this and use "`keep`" on it. (very :lost: even now)

At this point, I had already taken a look at the resolve docs. And well, I've started to add all the deps that weren't documented and resholve was finding :trophy: . That lead me to these issues in the way:

- "There is not yet a good way to resolve 'ping' in Nix builds." (Which don't come up with any suggestion, I've added it as an external)

- "timeout" had a "15" where the lore was expecting an "executable argument" (No clue how to say it to skip only this parameter, so I changed it to "cannot", which I'm pretty sure I should not)

- you repeat `"execer"` twice here: https://github.com/abathur/resholve/blob/f3faceb58d731e8d9f0dd244527007cf7a1fbfcc/docs/examples/lore.nix#L14 -- should be "wrapper" right? (This repeats in that README)

- "tmux" is executing its arguments, even worse, airgeddon uses "send-keys" to throw commands in it. (What should I do here? Again I went with "cannot")

- I had to come up with a hacky solution to keep optional dependencies optional, using the "external".

In the end, it ran, but the scripts in heredocs inside it were not using absolute references. They were even using the unpatched shebang. Airgeddon starts running one of these inside tmux and that wasn't finding the executables.